R/mwrite.R

```diff
@@ -186,6 +203,52 @@ mwrite_yaml <- function(x, file, digits = 8) {
   invisible(l)
 }
 
+#' Write a model to native mrgsolve format
+#' 
+#' Model code is written to a file in native mrgsolve format. This 
+#' can be useful for (1) breaking connection to NONMEM modeling outputs that 
+#' are imported by `$NMXML` or `$NMEXT` and (2) saving model updates (e.g., 
+#' an updated parameter list). Models can be read back using [mread()].
+#' 
+#' @inheritParams mwrite
+#' @inheritParams yaml_to_cpp
+#' 
+#' @details
+#' See important details in [mwrite_yaml()].
+#' 
+#' @return 
+#' A list containing data that was written out to the cpp file, with added 
+#' item `file`, is returned invisibly. 
+#' 
+#' @examples
+#' temp <- tempfile(fileext = ".mod")
+#' 
+#' mod <- modlib("pk1", compile = FALSE)
+#' 
+#' x <- mwrite_cpp(mod, file = temp)
+#' 
+#' mod <- mread(x$file, compile = FALSE)
+#' 
+#' mod
+#' 
+#' @seealso [mwrite_yaml()], [yaml_to_cpp()]
+#' 
+#' @md
+#' @rdname mwrite
+#' @export
+mwrite_cpp <- function(x, file, update = TRUE) {
```

Comment:
For this function, use the existing machinery to get mrgsolve model object to a list and write to temporary mrgsolve file, then copy that file to the requested output file / path.